x-pack/docs/en/security/es-security-principles.asciidoc

```diff
@@ -0,0 +1,55 @@
+[[es-security-principles]]
+== {es} security principles
+Protecting your {es} cluster and the data it contains is of utmost importance.
+There are numerous security practices and considerations for keeping your
+cluster safe, such as not saving passwords in plain text files or making sure
+that users have appropriate permissions (most users shouldn't be a superuser).
+These guidelines are important, but the following information provides something
+greater. These principles are foundational to running {es} in a secure manner.
+
+[discrete]
+[[security-dont-run-without]]
+=== Run {es} with security enabled
+Never run an {es} cluster without security enabled. This principle cannot be
+overstated. Running {es} without security leaves your cluster exposed to anyone
+on the internet, permitting anyone to download, modify, or delete any data in
```

Comment:
It leaves the cluster exposed to anyone who can hit ES packets. It's not the whole internet. Its still bad. Even if you trust everyone who can hit your production cluster you still want security in depth. And you probably shouldn't blindly trust everyone who can hit your production cluster.

Reply:
We could modify this to say:
>Running {es} without security leaves your cluster exposed to anyone who can access {es} packets, permitting these individuals to download, modify, or delete any data in your cluster.

Is "packets" the right term here? There are communications on the HTTP layer and on the transport layer, and I want to encapsulate both. Is "network traffic" more appropriate?

Reply:
"network traffic" is probably more right. Something like "anyone who send send network traffic to {es}"

"accessing packets" that makes me think of [wireshark](https://www.wireshark.org/). That's another whole kettle of fish. Probably not worth bringing up.

Mostly pedantic, but: in production folks that run ES are going to do it on a wired network. No "accessing packets" unless you own the hardware or the wires. If someone's up in your data center playing with your wires you are pretty well out of luck. Or the cloud equivalent. We've got TLS and you should use it, but if physical access crumples all of the depth we might have. Which is known. So its mostly not worth talking about.

One thing is semi-related is that ES only binds to localhost by default. This is because lots and lots of folks are starting it on their laptop. Folks like me hack ES locally, or write docs for it, or hack apps that talk to it. Or they slurp small data and use ES to slice it locally. Happens all the time. Laptops usually are on networks with fairly poor security. Public wifi. My house. localhost doesn't listen to anything outside of your laptop. That's good. Super good when `elastic:password` is a superuser.